### PR TITLE
Handle embedded MBQL snippets portably in Serdes v2

### PR DESCRIPTION
**Pull request by bshepherdson**
So far this is only supported for `Card.dataset_query`; it needs to be
extended to any other fields that hold MBQL as JSON.

Please help me build up a list of all the MBQL fields. A decent but imperfect
guide is to look for fields with interesting Toucan `:types`.

### CODE REVIEW

**Comment by diogormendes** [comment]
@bshepherdson , are we going to backport the new serdes as they got merged ?

File: enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj [hunk 1]
```diff
@@ -254,3 +255,91 @@
...
+          user1s     (atom nil)
+          db1d       (atom nil)
+          table1d    (atom nil)
+          field1d    (atom nil)
+          user1d     (atom nil)
+          card1d     (atom nil)
+          db2d       (atom nil)
+          table2d    (atom nil)
+          field2d    (atom nil)]
+
```
Comment (noahmoss) [inline enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj:280]: nit: extra line

**Review by noahmoss — APPROVED** [review]
Code all seems reasonable to me. I'm not sure what other fields have embedded MBQL though off the top of my head. Maybe @camsaul can help here.

**Comment by bshepherdson** [comment]
I've got a follow-up PR that does the same for `Metric.definition` and `Segment.definition`, and a similar transformation for `parameter_mappings` on `DashboardCard` and `Card`.

File: enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj (hunk 1 above)
Comment (bshepherdson) [reply enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj:280]: Submitting without this; the follow-up PR removes it.

**Pull request merged**